<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1268</width>
    <height>640</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1268</width>
    <height>640</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1268</width>
    <height>640</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QPushButton" name="btn_open_file">
    <property name="geometry">
     <rect>
      <x>1030</x>
      <y>10</y>
      <width>211</width>
      <height>41</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="text">
     <string>Open file</string>
    </property>
   </widget>
   <widget class="QSlider" name="zoom_slider">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>550</y>
      <width>201</width>
      <height>31</height>
     </rect>
    </property>
    <property name="minimum">
     <number>2</number>
    </property>
    <property name="maximum">
     <number>50</number>
    </property>
    <property name="value">
     <number>10</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="lbl_zoom">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>500</y>
      <width>200</width>
      <height>41</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>200</width>
      <height>0</height>
     </size>
    </property>
    <property name="text">
     <string>Zoom</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <widget class="QPushButton" name="btn_background_color">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>10</y>
      <width>201</width>
      <height>41</height>
     </rect>
    </property>
    <property name="maximumSize">
     <size>
      <width>16777215</width>
      <height>16777215</height>
     </size>
    </property>
    <property name="text">
     <string>Цвет фона</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>70</y>
      <width>231</width>
      <height>111</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="grid_output">
     <property name="sizeConstraint">
      <enum>QLayout::SetDefaultConstraint</enum>
     </property>
     <item row="3" column="0">
      <widget class="QLabel" name="label_vertex_count">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="text">
        <string> Вершины:</string>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QLabel" name="label_edge_output">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLabel" name="label_vertex_output">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="label_edge">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="text">
        <string> Ребра:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="label_namefile">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="text">
        <string> Имя файла:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLabel" name="label_namefile_output">
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>70</y>
      <width>201</width>
      <height>211</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="Model_grid">
     <item row="1" column="0">
      <widget class="QLabel" name="labe_axis_X">
       <property name="text">
        <string>Ось X</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0">
      <widget class="QSlider" name="slider_axis_Y">
       <property name="minimum">
        <number>-100</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="7" column="0">
      <widget class="QSlider" name="slider_axis_Z">
       <property name="minimum">
        <number>-100</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="6" column="0">
      <widget class="QLabel" name="label_axis_Z">
       <property name="text">
        <string>Ось Z</string>
       </property>
      </widget>
     </item>
     <item row="2" column="0">
      <widget class="QSlider" name="slider_axis_X">
       <property name="minimum">
        <number>-100</number>
       </property>
       <property name="maximum">
        <number>100</number>
       </property>
       <property name="pageStep">
        <number>5</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label_axis_Y">
       <property name="text">
        <string>Ось Y</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>270</y>
      <width>231</width>
      <height>140</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="grid_line">
     <item row="4" column="0" colspan="2">
      <widget class="QSlider" name="slider_line_thickness">
       <property name="focusPolicy">
        <enum>Qt::WheelFocus</enum>
       </property>
       <property name="minimum">
        <number>2</number>
       </property>
       <property name="maximum">
        <number>20</number>
       </property>
       <property name="pageStep">
        <number>2</number>
       </property>
       <property name="value">
        <number>2</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBothSides</enum>
       </property>
       <property name="tickInterval">
        <number>2</number>
       </property>
      </widget>
     </item>
     <item row="3" column="1" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_dotted_line">
       <property name="text">
        <string>Пунктир</string>
       </property>
      </widget>
     </item>
     <item row="5" column="0" colspan="2">
      <widget class="QPushButton" name="btn_line_color">
       <property name="text">
        <string>Цвет линий</string>
       </property>
      </widget>
     </item>
     <item row="3" column="0" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_solid_line">
       <property name="text">
        <string>Сплошная</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <widget class="QLabel" name="label_type_lines">
       <property name="text">
        <string> Тип линий</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>430</y>
      <width>231</width>
      <height>140</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="grid_vertex">
     <property name="topMargin">
      <number>0</number>
     </property>
     <property name="rightMargin">
      <number>0</number>
     </property>
     <property name="bottomMargin">
      <number>0</number>
     </property>
     <item row="3" column="0" colspan="3">
      <widget class="QSlider" name="slider_vertex_thickness">
       <property name="minimum">
        <number>5</number>
       </property>
       <property name="maximum">
        <number>40</number>
       </property>
       <property name="pageStep">
        <number>5</number>
       </property>
       <property name="value">
        <number>5</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="tickPosition">
        <enum>QSlider::TicksBothSides</enum>
       </property>
       <property name="tickInterval">
        <number>5</number>
       </property>
      </widget>
     </item>
     <item row="1" column="2" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_vertex_circle">
       <property name="text">
        <string>Круг</string>
       </property>
      </widget>
     </item>
     <item row="0" column="0" colspan="3">
      <widget class="QLabel" name="label_vertex">
       <property name="text">
        <string> Тип вершин</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="1" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_vertex_square">
       <property name="text">
        <string>Квадрат</string>
       </property>
      </widget>
     </item>
     <item row="1" column="0" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_vertex_null">
       <property name="text">
        <string>Пусто</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item row="4" column="0" colspan="3">
      <widget class="QPushButton" name="btn_vertex_color">
       <property name="text">
        <string>Цвет вершин</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>200</y>
      <width>232</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="grid_plane">
     <item row="0" column="0" colspan="2">
      <widget class="QLabel" name="label_type_plane">
       <property name="text">
        <string> Тип проекции </string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </item>
     <item row="1" column="0" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_plane_parall">
       <property name="text">
        <string>Параллельная</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1" alignment="Qt::AlignHCenter">
      <widget class="QRadioButton" name="btn_radio_plane_center">
       <property name="text">
        <string>Центральная</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="s21_OpenGLWidget" name="openGLWidget">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>0</y>
      <width>771</width>
      <height>581</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="Frame_line" native="true">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>270</y>
      <width>232</width>
      <height>140</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
border: 2px solid #C0C0FF;
border-radius: 10px;</string>
    </property>
   </widget>
   <widget class="QWidget" name="Frame_vertex" native="true">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>430</y>
      <width>232</width>
      <height>141</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
border: 2px solid #C0C0FF;
border-radius: 10px;</string>
    </property>
   </widget>
   <widget class="QWidget" name="Frame_pane" native="true">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>200</y>
      <width>231</width>
      <height>55</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
border: 2px solid #C0C0FF;
border-radius: 10px;</string>
    </property>
   </widget>
   <widget class="QWidget" name="Frame_output" native="true">
    <property name="geometry">
     <rect>
      <x>1020</x>
      <y>70</y>
      <width>231</width>
      <height>111</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">
border: 2px solid #C0C0FF;
border-radius: 10px;</string>
    </property>
   </widget>
   <widget class="QWidget" name="gridLayoutWidget_6">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>290</y>
      <width>201</width>
      <height>201</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="Model_grid_2">
     <item row="3" column="5">
      <widget class="QLabel" name="label_angel_Y">
       <property name="text">
        <string>Угол Y</string>
       </property>
      </widget>
     </item>
     <item row="1" column="5">
      <widget class="QLabel" name="labe_angel_X">
       <property name="text">
        <string>Угол X</string>
       </property>
      </widget>
     </item>
     <item row="7" column="5">
      <widget class="QSlider" name="slider_angel_Z">
       <property name="minimum">
        <number>-160</number>
       </property>
       <property name="maximum">
        <number>160</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="6" column="5">
      <widget class="QLabel" name="label_angel_Z">
       <property name="text">
        <string>Угол Z</string>
       </property>
      </widget>
     </item>
     <item row="5" column="5">
      <widget class="QSlider" name="slider_angel_Y">
       <property name="minimum">
        <number>-160</number>
       </property>
       <property name="maximum">
        <number>160</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item row="2" column="5">
      <widget class="QSlider" name="slider_angel_X">
       <property name="minimum">
        <number>-160</number>
       </property>
       <property name="maximum">
        <number>160</number>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="">
    <property name="geometry">
     <rect>
      <x>240</x>
      <y>590</y>
      <width>771</width>
      <height>32</height>
     </rect>
    </property>
    <layout class="QGridLayout" name="grid_save">
     <item row="0" column="0">
      <widget class="QPushButton" name="btn_save_jpeg">
       <property name="text">
        <string>Save .jpeg</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QPushButton" name="btn_save_bmp">
       <property name="text">
        <string>Save .bmp</string>
       </property>
      </widget>
     </item>
     <item row="0" column="2">
      <widget class="QPushButton" name="btn_save_gif">
       <property name="text">
        <string>Save .gif</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <zorder>Frame_vertex</zorder>
   <zorder>Frame_pane</zorder>
   <zorder>Frame_output</zorder>
   <zorder>Frame_line</zorder>
   <zorder>btn_open_file</zorder>
   <zorder>zoom_slider</zorder>
   <zorder>lbl_zoom</zorder>
   <zorder>btn_background_color</zorder>
   <zorder>gridLayoutWidget</zorder>
   <zorder>gridLayoutWidget_2</zorder>
   <zorder>gridLayoutWidget_3</zorder>
   <zorder>gridLayoutWidget_4</zorder>
   <zorder>gridLayoutWidget_5</zorder>
   <zorder>btn_save_jpeg</zorder>
   <zorder>openGLWidget</zorder>
   <zorder>gridLayoutWidget_6</zorder>
   <zorder>btn_save_bmp</zorder>
   <zorder>btn_save_gif</zorder>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>s21_OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>s21_openglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
